sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -832,7 +798,21 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                 if ((ex as ServiceBusException)?.Reason == ServiceBusException.FailureReason.SessionLockLost)
                 {
                     ServiceBusSessionReceiver sessionReceiver = (ServiceBusSessionReceiver)receiver;
-                    sessionIdsReceiverMap.TryRemove(sessionReceiver.SessionId, out var _);
+                    if (sessionIdsReceiverMap.ContainsKey(sessionReceiver.SessionId))
```

Comment:
Why do we do the containsKey check now?

Reply:
I added this check because I'm thinking this error can come even if we don't specify the sessionIds. 

I can replace this with `sessionId != null` check.

Reply:
Yeah that would be a bit more clear.